src/drivers/intel/hda/hda-dma.c

```diff
@@ -860,9 +860,13 @@ static int hda_dma_link_check_xrun(struct dma_chan_data *chan)
 	if (chan->direction == DMA_DIR_MEM_TO_DEV && dgcs & DGCS_BUR) {
 		tr_err(&hdma_tr, "hda_dma_link_check_xrun(): underrun detected");
 		dma_chan_reg_update_bits(chan, DGCS, DGCS_BUR, DGCS_BUR);
-	} else if (chan->direction == DMA_DIR_DEV_TO_MEM && dgcs & DGCS_BOR) {
+		return -ENODATA;
+	}
+
+	if (chan->direction == DMA_DIR_DEV_TO_MEM && dgcs & DGCS_BOR) {
```

Comment:
nit-pick: I would add parentheses here and line 860, this sort of code is awful.